lib/util/make-knex.js

```diff
@@ -36,17 +36,32 @@ function initContext(knexFn) {
     // If container is provided, returns a promise for when the transaction is resolved.
     // If container is not provided, returns a promise with a transaction that is resolved
     // when transaction is ready to be used.
-    transaction(container, config) {
-      const trx = this.client.transaction(container, config);
-      trx.userParams = this.userParams;
+    transaction(container, _config) {
+      // Shallow copy _config to avoid side-effects
+      const config = Object.assign({}, _config);
+
+      // Backwards-compatibility: the default value of `config.doNotRejectOnRollback`
+      // changes depending upon whether or not a `container` was provided.
+      if(isUndefined(config.doNotRejectOnRollback)) {
+        config.doNotRejectOnRollback = !container;
+      }
 
-      if (container) {
+      if(container) {
+        const trx = this.client.transaction(container, config);
+        trx.userParams = this.userParams;
         return trx;
+      } else {
+        const initPromise = new Promise((resolve, reject) => {
+          _resolve = resolve;
+          _reject = reject;
+        });
+
+        const trx = this.client.transaction(_resolve, config);
```

Comment:
maybe move this lines inside Promise constructor?

Reply:
Oh -- you're right!  I didn't even notice that the refactoring allowed for the code to be cleaned up in that way.  Neat!

(I'll push the change once the tests finish running)

Reply:
also _resolve and _reject wasn't declared, but this will be fixed by that move

Reply:
Oops -- looks like I deleted the `let` statement for those when I was rearranging the code earlier.  Anyhoo: the latest commit moves everything inside of the Promise constructor.